File: VB/Export_Import_Example/IsolatedStorageStreamProvider.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports System.IO
Imports DevExpress.Office.Services
Imports System.IO.IsolatedStorage

Namespace Export_Import_Example
	#Region "#streamprovider"
	Public Class IsolatedStorageStreamProvider
		Implements IUriStreamProvider
		Private sourceUri As String

		Public Sub New(ByVal sourceUri As String)
			Me.sourceUri = sourceUri
		End Sub

		#Region "IUriStreamProvider Members"
		Private Function GetStream(ByVal uri As String) As Stream Implements IUriStreamProvider.GetStream 
			If uri.ToLower().StartsWith("http") Then
				Return Nothing
			End If
			Dim path As String = sourceUri & uri.Replace("/", "\")
			Using store As IsolatedStorageFile = IsolatedStorageFile.GetUserStoreForApplication()
				Return store.OpenFile(path, FileMode.Open)
			End Using
		End Function
		#End Region
	End Class
	#End Region ' #streamprovider
End Namespace

File: CS/Export_Import_Example/IsolatedStorageStreamProvider.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using System.IO;
using DevExpress.XtraRichEdit.Services;
using System.IO.IsolatedStorage;

namespace Export_Import_Example
{
    #region #streamprovider
    public class IsolatedStorageStreamProvider : IUriStreamProvider
    {
        string sourceUri;

        public IsolatedStorageStreamProvider(string sourceUri)
        {
            this.sourceUri = sourceUri;
        }

        #region IUriStreamProvider Members
        Stream IUriStreamProvider.GetStream(string uri)
        {
            if (uri.ToLower().StartsWith("http"))
                return null;
            string path = sourceUri + uri.Replace("/", "\\");
            using (IsolatedStorageFile store = IsolatedStorageFile.GetUserStoreForApplication()) {
                return store.OpenFile(path, FileMode.Open);
            }
        }
        #endregion
    }
    #endregion #streamprovider
}

